x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curations_logic.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+
+import {
+  clearFlashMessages,
+  setSuccessMessage,
+  flashAPIErrors,
+} from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { DELETE_MESSAGE, SUCCESS_MESSAGE } from './constants';
+import { Curation, CurationsAPIResponse } from './types';
+
+interface CurationsValues {
+  dataLoading: boolean;
+  curations: Curation[];
+  meta: Meta;
+}
+
+interface CurationsActions {
+  onCurationsLoad(response: CurationsAPIResponse): CurationsAPIResponse;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  loadCurations(): void;
+  deleteCurationSet(id: string): string;
+}
+
+export const CurationsLogic = kea<MakeLogicType<CurationsValues, CurationsActions>>({
+  path: ['enterprise_search', 'app_search', 'curations_logic'],
+  actions: () => ({
+    onCurationsLoad: ({ results, meta }) => ({ results, meta }),
+    onPaginate: (newPageIndex) => ({ newPageIndex }),
+    loadCurations: true,
+    deleteCurationSet: (id) => id,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        loadCurations: () => true,
+        onCurationsLoad: () => false,
+      },
+    ],
+    curations: [
+      [],
+      {
+        onCurationsLoad: (_, { results }) => results,
+      },
+    ],
+    meta: [
+      DEFAULT_META,
```

Comment:
I want to note this as another change from the standalone UI.

Currently, the standalone UI defaults to `{}` (meaning it falls back to whatever page size specified/defaulted by the API route), but I figured: we have this constant, why not use it?

We have way inconsistent/varying pagination sizes across the App Search UI right now. It's currently 25 for Curations, 10 for our engines tables, 25 for credentials... I would really like to standardize this across the board to make our UI feel more predictable, and I think `DEFAULT_META` is an excellent way of accomplishing that (without having to go in and make deprecation changes to our core APIs), and that 10 is generally a saner standard/default.

Thoughts?

Reply:
I think this is great. This also makes the typing of `meta` here easier, because `meta.page`, etc. will always be there, whether or not the data has been fetched yet or not, so you can just type it as `Meta`.

Reply:
I think having a default pagination is a good idea, if we want to use something specific for a view we'll always have that option